src/commands/config/set.command.ts

```diff
@@ -0,0 +1,27 @@
+import { GlobalCommandOptions } from "../GlobalCommandOptions.ts";
+import { TopLevelCommand } from "../TopLevelCommand.ts";
+import { CollieConfig } from "../../model/CollieConfig.ts";
+import { Logger } from "../../cli/Logger.ts";
+import { CollieRepository } from "../../model/CollieRepository.ts";
+import { InteractivePrompts } from "../interactive/InteractivePrompts.ts";
+import { CLI } from "../../info.ts";
+
+export function registerSetCmd(program: TopLevelCommand) {
+  program
+    .command("set-foundation [foundation:foundation]")
+    .description("Set the foundation config property.")
+    .example(
+      "Set foundation `myfoundation` in repository config.",
+      `${CLI} config set-foundation myfoundation`,
+    )
+    .action(
+      async (opts: GlobalCommandOptions, foundationArg: string | undefined) => {
+        const collieRepo = await CollieRepository.load();
+        const logger = new Logger(collieRepo, opts);
+        const foundation = foundationArg ||
+          (await InteractivePrompts.selectFoundation(collieRepo, logger));
+
+        CollieConfig.setFoundation(foundation, logger);
```

Comment:
d: should this be a static class? or should we rather put this into `CollieRepository` where we already have an async load operation available to load the current config.

Reply:
CollieConfig is now a proper class with `CollieRepository` as a dependency.